source/vm.cc

```diff
@@ -82,6 +82,17 @@ namespace phoenix {
 		_m_temporary_strings = add_temporary_strings_symbol();
 	}
 
+	std::shared_ptr<instance> vm::init_opaque_instance(symbol* sym) {
+		auto cls = sym;
+		if (cls->type() == datatype::instance) {
+			cls = find_symbol_by_index(cls->parent());
+		}
```

Comment:
This will fail if the instance inherits from a prototype. We need to check all parents instead, until we find a class.